src/wkt/src/message.rs

```diff
@@ -71,28 +69,44 @@ where
     }
 }
 
-pub(crate) fn to_json_string<T>(message: &T) -> Result<Map, Error>
+/// Write the serialization of `T` into the `value` field of a map.
+///
+/// This is used for types where the JSON serialization of `T` is not an
+/// object (e.g. bool, number, string, etc.). The encoding is a map, and these
+/// values need a key.
+///
+/// It is also used for `Any`. Flatly serializing an `Any` would have
+/// conflicting `@type` fields.
```

Comment:
I think the motivation is to match the ProtoJSON serialization which defines special encodings for Duration, Timestamp, Any, and the `*Value` types (which look like `bool`, `number`, etc. in Rust).